concepts/pattern-matching/about.md

```diff
@@ -0,0 +1,52 @@
+# Introduction
+
+While `if/else` expressions can be used to execute conditional logic, Haskell also has a  more powerful way to execute conditional logic: pattern matching.
+With pattern matching, a value can be tested against one or more _patterns_.
+An example of such a pattern is the _constant pattern_, which matches a value against a constant (e.g. `1` or `"hello"`).
+
+When defining functions, you can define separate function bodies for different patterns.
+This leads to clean code that is simple and readable.
+You can pattern match on any data type — numbers, characters, lists, tuples, etc.
+
+For example, a trivial function that checks if the number supplied to it is a `1` or `0` could be expressed like this:
+
+```haskell
+describe :: Int -> String
+describe 0 = "Zero"
+describe 1 = "One"
+```
+
+Pattern matching starts to shine when used together with other patterns, for example the _variable pattern_:
+
+```haskell
+describe :: Int -> String
+describe 0 = "Zero"
+describe n = "Not zero"
```

Comment:
I wonder if it would be better to use an example that could make use of the `n` here. Is that not important?

Reply:
Not a direct answer to your question, but: we're still talking about it on Slack.

Reply:
The only way I can think of using the `n` is something like:
```
describe n = show n ++ " is not zero
```
But we haven't introduced `show` or `++` yet. I think not using the variable is preferable to using functions not previously introduced. Also, that example is an intermediate step to the next one where `n` is used for the conditional. Thoughts?

Reply:
I agree with @pwadsworth's reasoning here. I had the same thought as @petertseng initially, but then I read on and saw that it was used in the next example, so I'm fine with it. And yes, using `show` or `++` is not advisable.

Reply:
Given that `show` and `++` are inappropriate, that implies that the result of the function should be something other than String, rather than discarding the idea of using the `n` entirely!

For example, one could present a function that brings a number closer to zero by subtracting 1 from it, if it isn't already zero:

```haskell
closerToZero 0 = 0
closerToZero 1 = 0
closerToZero 2 = 1
```

and then, one could introduce the ability to bind, replacing the 1 and 2 cases

```haskell
closerToZero 0 = 0
closerToZero n = n - 1
```

Then, guards could be used to say that negative numbers should instead have 1 added to them, because that's the way to bring them closer to zero.

This doesn't necessarily have to be the idea that is actually used, but you must know that it is my duty to ensure that the idea is considered, and when I see no evidence that the idea has been considered I am required to bring it up.

Reply:
I can get onboard with that. It makes use of the variable and doesn't distract from the point of patter matching. @ErikSchierboom / @kytrinyx  do you agree ?

Reply:
I really like the `closerToZero` suggestion. I'd say ship it!